### PR TITLE
 [jit] Convert functional dropouts to weak script

### PR DESCRIPTION
**Pull request by driazati**
To convert `nn.functional.dropout`
* `_VF` had to be exposed as a Python module so this PR adds a module class to forward to `torch._C._VariableFunctions`
* rng state between calls in the tests needed to be made consistent

### CODE REVIEW

File: torch/jit/__init__.py [hunk 1]
```diff
@@ -1312,6 +1312,11 @@ def forward(self, input):
     'pairwise_distance',
     'prelu',
     'hardshrink',
+    'dropout',
```
Comment (wanchaol) [inline torch/jit/__init__.py]: It looks like we have dropout/alpha_dropout/feature_alpha_dropout as Aten ops, why we are adding them to the blacklist?

Comment (driazati) [reply torch/jit/__init__.py]: When these are used from `torch.nn.functional` they should have the same semantics as the Python versions (which is mostly throwing exceptions on bad input / forwarding different args), so this makes them use the weak script versions from `torch.nn.functional` rather than just the aten op

Comment (wanchaol) [reply torch/jit/__init__.py]: gotcha, shall we make the comment above be more clear so that we are not confused with it?

File: torch/nn/_VF.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import torch
+import sys
+import types
+
+
+class VFModule(types.ModuleType):
```
Comment (zdevito) [inline torch/nn/_VF.py]: I think this is a common pattern now where there is a module that wraps an object. Can this be turned into a utility class and used in both places?

File: torch/nn/functional.py [hunk 3]
```diff
@@ -621,26 +623,36 @@ def dropout(input, p=0.5, training=True, inplace=False):
         training: apply dropout if is ``True``. Defualt: ``True``
         inplace: If set to ``True``, will do this operation in-place. Default: ``False``
     """
-    if p < 0 or p > 1:
+    if p < 0. or p > 1.:
         raise ValueError("dropout probability has to be between 0 and 1, "
                          "but got {}".format(p))
-    f = _VF.dropout_ if inplace else _VF.dropout
-    return f(input, p, training)
+    if inplace:
```
Comment (zdevito) [inline torch/nn/functional.py]: does script not support ternary if? I thought it did.

**Review by zdevito — APPROVED** [review]
Looks good. If ternary works in script we should use it. If not, we should add it to the features we want to add at p1.

File: torch/nn/_VF.py (hunk 2 above)
Comment (driazati) [reply torch/nn/_VF.py]: Will do in a followup PR
